src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -855,6 +855,10 @@ namespace winrt::TerminalApp::implementation
         newTabFlyout.Opening([this](auto&&, auto&&) {
             _FocusCurrentTab(true);
         });
+        // Necessary for fly-out sub items to get focus on a tab before collapsing. Related to #15049
+        newTabFlyout.Closing([this](auto&&, auto&&) {
+            _FocusCurrentTab(true);
```

Comment:
@carlos-zamora, can you do an accessibility pass over this?

Reply:
Works great!